docs/src/variables.md

```diff
@@ -580,6 +587,47 @@ julia> x = @variable(model, [i=1:2], base_name="x", lower_bound=i, integer=true)
     Creating two named JuMP variables with the same name results in an error at
     runtime. Use anonymous variables as an alternative.
 
+## Constrained variables
+
+To create variables constrained to belong to the [`SecondOrderCone`](@ref) either
+of the following two methods can be used:
+```jldoctest constrained_variables; setup=:(model=Model())
+julia> @variable(model, x[1:3] in SecondOrderCone())
+3-element Array{VariableRef,1}:
+ x[1]
+ x[2]
+ x[3]
+
+julia> @variable(model, y[1:3])
+3-element Array{VariableRef,1}:
+ y[1]
+ y[2]
+ y[3]
+
+julia> @constraint(model, y in SecondOrderCone())
+[y[1], y[2], y[3]] ∈ MathOptInterface.SecondOrderCone(3)
+```
+The variables `x` in the example above are called *constrained variables* in
+contrast which the variables `y` which are called *free variables* because
+the constraint added on the variables is added after their creation as free
```

Comment:
This is still not quite clear. Variable bounds and integrality constraints are specified when variables are created, but they don't involve constrained variables, unless you explicitly use `in` or `set=`.